packages/react-stand-in/src/createClassProxy.js

```diff
@@ -40,18 +40,14 @@ function createClassProxy(InitialComponent, proxyKey, wrapResult = identity) {
     lastInstance = this
   }
 
-  let hasInitialResult = false
-  let initialResult = null
-
   function proxiedRender() {
     inject(this, proxyGeneration, injectedMembers)
 
     let result
 
-    if (hasInitialResult) {
-      result = initialResult
-      hasInitialResult = false
-      initialResult = null
+    if (this[CACHED_RESULT]) {
```

Comment:
You may need to do an explicit `hasOwnProperty` check here. Imagine if the functional component returns `null` or something – then this check will fail. That's why I had the explicit `hasInitialResult` earlier.

Reply:
Nope. Result have to be an instance with a render function to trigger this code branch.

Reply:
Sorry, I may not have explained correctly in the comments. This code path only gets hit for normal functional components. When the proxies component actually returns a React instance, we don’t hit this code path. The initial result is instead the initial return value from a normal functional component.

Reply:
See https://github.com/gaearon/react-hot-loader/commit/f27db13ffbfd8f2ab753b94a6384a2b21ca55146. This affected the test results for SFCs.

Reply:
You are right. I've `inverted` the expectations.

Reply:
I can PR a fix and add a regression test.

Reply:
It will be awesome. I’ve stuck with home duties.